internal/testutil/datagen/datagen.sh

```diff
@@ -0,0 +1,250 @@
+
+    echo "Extracting kubelet /stats/summary"
+    mkdir -p kubelet/stats
+    kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape kubelet stats/summary > kubelet/stats/summary
+
+    # KSM endpoint
+    mkdir -p ksm
+    echo "Extracting ksm /metrics"
+    kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape ksm > ksm/metrics
+
+    if [[ "$DISABLE_CONTROLPLANE" != "1" ]]; then
+        # Control plane components
+        mkdir -p controlplane/api-server
+        echo "Extracting api-server /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane apiserver > controlplane/api-server/metrics
+
+        mkdir -p controlplane/etcd
+        echo "Extracting etcd /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane etcd > controlplane/etcd/metrics
+
+        mkdir -p controlplane/controller-manager
+        echo "Extracting controller-manager /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane controllermanager > controlplane/controller-manager/metrics
+
+        mkdir -p controlplane/scheduler
+        echo "Extracting scheduler /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane scheduler > controlplane/scheduler/metrics
+    else
+        echo "Skipping control plane metrics"
+    fi
+
+    # K8s objects
+    echo "Generating list of kubernetes resources"
+    kubedump nodes
+    kubedump namespaces
+    kubedump endpoints
+    kubedump services
+    kubedump pods
+
+    cd -
+    if [[ "$DISABLE_CLEANUP" != "1" ]]; then
+      cleanup
+    fi
+}
+
+# bootstrap suvcommand installs the required components in the cluster to generate the testdata.
+# If $SKIP_INSTALL is non-empty, it skips deploying KSM, the dummy resources, and the scrapper pod and just copies
+# script inside an scraper pod that is assumed to exist already.
+function bootstrap() {
+    echo -e "Using context $(kubectl config current-context), is this ok?\n^C now if it is not."
+    read
+
+    if [[ -z $SKIP_INSTALL ]]; then
+        echo "Installing e2e-resources chart"
+        if [[ "$IS_MINIKUBE" = "1" ]]; then
+            # Enable PVC if we are in minikube
+            minikube_args="--set persistentVolumeClaim.enabled=true --set loadBalancerService.fakeIP=127.1.2.3"
+            minikube addons enable metrics-server
+        fi
+
+        helm dependency update ../../../e2e/charts/e2e-resources > /dev/null
+        helm upgrade --install e2e ../../../e2e/charts/e2e-resources -n $scrapper_namespace --create-namespace \
+          --set scraper.enabled=true \
+          --set persistentVolume.enabled=true \
+          $minikube_args \
+          $HELM_E2E_ARGS
+
+        echo "Installing KSM"
+        helm dependency update ../../../e2e/charts/ksm > /dev/null
+        helm upgrade --install ksm ../../../e2e/charts/ksm -n ksm --create-namespace --wait
```

Comment:
I would change this name to avoid collision with other deployment, testing it locally I got:
```
Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "ksm-kube-state-metrics" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-namespace" must equal "ksm": current value is "kube-system"

```

Reply:
Did you have a release also named `ksm` in `kube-system`? 😆 
Makes sense to me, I'll change it to `scraper-ksm` for example.

Reply:
I had it in a different namespace, but cluster resources are overlapping 😖 
if it becomes part of the e2e resources it is not an issue anymore 😄

Reply:
Done in [`a4dcdbe` (#269)](https://github.com/newrelic/nri-kubernetes/pull/269/commits/a4dcdbed6b90ed85dd53697caec64451f0db2b40) :)